code/datums/mapgen/CaveGenerator.dm

```diff
@@ -4,6 +4,8 @@
 	var/open_turf_types = list(/turf/open/floor/plating/asteroid = 1)
 	///Weighted list of the types that spawns if the turf is closed
 	var/closed_turf_types =  list(/turf/closed/mineral/random/volcanic = 1)
+	///List of turf types (subtypes included) to skip when generating terrain
+	var/blacklisted_turf_types
```

Comment:
static

Reply:
Is static separate on every subtype? I made this work like the other vars, in that each generator can define its own turf types to ignore, so as to avoid incompatibility with other cave generators ported from other codebases.

This might even be replaced entirely if/when tgstation implements a better solution for not overriding lava rivers tbh.

Reply:
if you generate different static value on subtypes it should be separate

Reply:
No which is very annoying, so you can't make it static which is dumb. As long as the list is cleared after use, it shouldn't be a problem